### DISCUSSION TITLE
About mobile cookie in FriendlyUrlController.cs

### DISCUSSION THREAD

**Discussion by thienvc** [dnnsoftware/Dnn.Platform#6003] [General]
When identifying browser type and mobile, in the following code

https://github.com/dnnsoftware/Dnn.Platform/blob/949cac213d78038539a8359c34667802c6d907ab/DNN%20Platform/Library/Entities/Urls/FriendlyUrlController.cs#L837-L882

I wonder why cookies are not used in the request, but in the response
`HttpCookie viewMobileCookie = response.Cookies[MobileViewSiteCookieName];`
`HttpCookie viewMobileCookie = request.Cookies[MobileViewSiteCookieName];`
What is the difference?

**Comment by mitchelsellers** [comment]
My assumption is that the mobile identification would have been done earlier in the pipeline, and you cannot change the incoming request cookies, but you can change the outgoing response cookies, so that is the "most current" value.

**Reply by thienvc** [reply]
I think differently, if it is as you say, why do we need to call the isMobile = IsMobileClient() function?

**Reply by mitchelsellers** [reply]
That is doing the browser dectection, if the cookie was already set to identify it, no need to do it again, and there are some funny order of operations with regards to identification & handlers.  (Just my best guess at this point).

**Reply by thienvc** [reply]
You can debug again to know clearly, `HttpCookie viewMobileCookie = response.Cookies[MobileViewSiteCookieName];` then `viewMobileCookie` will not be NULL at any time.
That leads to the statement `if (viewMobileCookie == null)`  which will always be False (Algorithmically wrong)

**Comment by thienvc** [comment]
![image](https://github.com/dnnsoftware/Dnn.Platform/assets/1115393/566efadc-a920-4cd3-8223-b2bb627d32eb)
Can someone explain why every request, whether it is the first or subsequent times, has these cookies sent back to the browser?
Is this really necessary except for the first request!?
For example, in the browser the cookie dnn_Mobile=true already exists, so do subsequent requests need to continue sending down a similar cookie with the same value?
